lib/cf-socket.c

```diff
@@ -1707,7 +1726,8 @@ static CURLcode cf_udp_setup_quic(struct Curl_cfilter *cf,
               ctx->sock, ctx->ip.local_ip, ctx->ip.local_port,
               ctx->ip.remote_ip, ctx->ip.remote_port);
 
-  (void)curlx_nonblock(ctx->sock, TRUE);
+  if(blocking)
+    (void)curlx_nonblock(ctx->sock, TRUE);
```

Comment:
Currently it is always `FALSE` because the only caller of `cf_udp_setup_quic()` is `cf_udp_connect()` that passes the non-blocking socket to it, so it would be safe to eliminate this `curlx_nonblock()` in `cf_udp_setup_quic()` at this moment, but I prefer to retain this parameter in case `cf_udp_setup_quic()` gets called elsewhere. Or do you think it's needless?

Reply:
I think it is needless, and just risks getting "complaints" by static analyzers finding dead code. I think you can rather add a brief comment mentioning this instead.